cpp/pybind/io/class_io.cpp

```diff
@@ -194,22 +194,65 @@ void pybind_class_io(py::module &m_io) {
     docstring::FunctionDocInject(m_io, "read_point_cloud",
                                  map_shared_argument_docstrings);
 
+    m_io.def(
+            "read_point_cloud_from_bytes",
+            [](const py::bytes &bytes, const std::string &format,
+               bool remove_nan_points, bool remove_infinite_points,
+               bool print_progress) {
+                const char *dataptr = PYBIND11_BYTES_AS_STRING(bytes.ptr());
+                auto length = PYBIND11_BYTES_SIZE(bytes.ptr());
+                auto data = std::vector<char>(dataptr, dataptr + length);
```

Comment:
I think using py::bytes requires holding the GIL. I can go with a raw buffer but we'd still have to copy to avoid issues in concurrent code

Reply:
I made the switch in 1984a20, but there will still be a copy :(

Reply:
From my own testing, most of the time spent was actually in `utility::FormatString` (for write) and `std::getline` and `sscanf` (for read). It seems sscanf will be slow on glibc calls strlen on every input (musl libc doesn't have this issue). I added an `utility::FastFormatString` which showed a remarkable speed improvement using `asprintf` instead of `snprintf`, but the downside is that it won't be available on Windows. I didn't attempt to get `std::getline` to be faster, but I'm sure there are things we could do.

On a 30mb xyz file:
```
Read From Bytes took: 1.59355s. <-- ~1s spent in std::getline
Read To File took: 0.595679s.
Write From Bytes took: 0.811818s.
Write To File took: 0.49677s.
```

On a 4k xyz file:
```
Read From Bytes took: 0.00023s.
Read To File took: 0.000512s.
Write From Bytes took: 0.000118s.
Write To File took: 0.000271s.
```

Reply:
Forgot to mention that the File I/O up there was w/ an SSD. Results might be very different on HDD's.